pkg/scheduler/schedule_one.go

```diff
@@ -486,14 +485,10 @@ func (sched *Scheduler) findNodesThatFitPod(ctx context.Context, fwk framework.F
 	nodes := allNodes
 	if !preRes.AllNodes() {
 		nodes = make([]*framework.NodeInfo, 0, len(preRes.NodeNames))
-		for _, n := range allNodes {
-			if !preRes.NodeNames.Has(n.Node().Name) {
-				// We consider Nodes that are filtered out by PreFilterResult as rejected via UnschedulableAndUnresolvable.
-				// We have to record them in NodeToStatusMap so that they won't be considered as candidates in the preemption.
-				diagnosis.NodeToStatusMap[n.Node().Name] = framework.NewStatus(framework.UnschedulableAndUnresolvable, "node is filtered out by the prefilter result")
-				continue
+		for nodeName := range preRes.NodeNames {
+			if nodeInfo, err := sched.nodeInfoSnapshot.Get(nodeName); err == nil {
```

Comment:
Add a comment that `preRes.NodeNames` could potentially return Node names that don't actually exist.

Reply:
Is it possible to keep the logic `!preRes.NodeNames.Has(n.Node().Name)`, but not put the UnschedulableAndUnresolvable nodes into NodeToStatusMap,  just continue, so as to avoid Node names that don't actually exist. At the same time, I think the number of nodes in preRes will not be very large, so it will not cause much performance loss

Reply:
No, iterating over the small number of preres nodes and checking if they actually exist is far more efficient than iterating over all nodes and checking if they are in the ver small set of preres node names.

 The `!preRes.NodeNames.Has(n.Node().Name)` check shows up in profiles with multi-thousand node runs as a significant contributor to time in this spot

Reply:
> No, iterating over the small number of preres nodes and checking if they actually exist is far more efficient than iterating over all nodes and checking if they are in the ver small set of preres node names.
> 
> The `!preRes.NodeNames.Has(n.Node().Name)` check shows up in profiles with multi-thousand node runs as a significant contributor to time in this spot

Yes, and it also avoids the problem of #123465

Reply:
In addition, can benchmark #124728 be merged into the Master? Because when facing this kind of problem, the local test can be quickly performed to check the change in throughput.

Reply:
If a benchmark is not part of the CI, it will eventually fall into oblivion. We need tests in `test/integration/scheduler_perf`, but we can leave it for a follow up.

Reply:
> If a benchmark is not part of the CI, it will eventually fall into oblivion. We need tests in `test/integration/scheduler_perf`, but we can leave it for a follow up.



> If a benchmark is not part of the CI, it will eventually fall into oblivion. We need tests in `test/integration/scheduler_perf`, but we can leave it for a follow up.

haha, i see

Reply:
> Is it possible to avoid Node names that don't actually exist.

Prefilter only looks at the pod spec, not existing nodes, so whatever limiting node names exist in the pod spec are what prefilter returns. This can unavoidably include node names that do not exist.

Ensuring `sched.nodeInfoSnapshot.Get(nodeName)` returns the info for the node is how we avoid propagating those non-existent nodes further.

Reply:
> Ensuring `sched.nodeInfoSnapshot.Get(nodeName)` returns the info for the node is how we avoid propagating those non-existent nodes further.

Oh, that's fine. The original comment just said that preFilter may have potentially non-existent nodes. `sched.nodeInfoSnapshot.Get(nodeName)` can indeed avoid non-existent nodes

Reply:
Chipping in here. There's a problem with the volume-binding prefilter. It assumes that the matchConstraints hostname label is identical to the node's name - it does not have to be. See #125336 for a walkthrough of how ths goes wrong.

Reply:
Looking at the issue, it's not relevant to this PR, but rather #109877.